tests/suite/graceful_recovery_test.go

```diff
@@ -370,7 +370,12 @@ func checkContainerLogsForErrors(ngfPodName string, checkNginxLogsOnly bool) {
 			// FIXME(salonichf5) remove this error message check
 			// when https://github.com/nginxinc/nginx-gateway-fabric/issues/2090 is completed.
 			expectedError2 := "no live upstreams while connecting to upstream"
-			Expect(line).To(Or(ContainSubstring(expectedError1), ContainSubstring(expectedError2)))
+			expectedError3 := "no resolver defined to resolve"
```

Comment:
this error occurs when ngf is restarted, we are polling curl to tea and coffee application to check for working traffic. 
I am confused on how to handle this cause this sets bad practice(adding more error messages to avoid). 

@kate-osborn Wondering if you know why it would not throw the `no live upstreams while connecting to upstream` if we have prefixed `upstream_` to the `proxy_pass` , `upstream` and `split-client` names.

Is NGINX always expecting backend names to be in a certain format?